src/legacy/core_plugins/data/public/search/search_bar/lib/saved_query_service.ts

```diff
@@ -89,24 +94,32 @@ export const createSavedQueryService = (
 
     return parseSavedQueryObject(rawQueryResponse);
   };
-
+  // we have to tell the saved objects client how many to fetch, otherwise it defaults to fetching 20 per page
   const getAllSavedQueries = async (): Promise<SavedQuery[]> => {
```

Comment:
Since we're now fetching the total count internally, I've removed the perPage and page parameters from the function signature. `getAllSavedQueries` now truly returns all of them.